src/command_modules/azure-cli-acr/azure/cli/command_modules/acr/_validators.py

```diff
@@ -19,3 +20,29 @@ def validate_registry_name(namespace):
 
         if not result.name_available: #pylint: disable=no-member
             raise CLIError(result.message) #pylint: disable=no-member
+
+def validate_headers(namespace):
+    ''' Extracts multiple space-separated headers in key[=value] format '''
+    if isinstance(namespace.headers, list):
+        headers_dict = {}
+        for item in namespace.headers:
+            headers_dict.update(validate_header(item))
+        namespace.headers = headers_dict
+
+def validate_header(string):
+    ''' Extracts a single header in key[=value] format '''
+    result = {}
+    if string:
+        comps = string.split('=', 1)
+        result = {comps[0]: comps[1]} if len(comps) > 1 else {string: ''}
+    return result
+
+def validate_replication_location(namespace):
+    if namespace.location:
+        registry_location, _ = get_registry_location_by_name(
+            namespace.registry_name, namespace.resource_group_name)
+        if namespace.location == "".join(registry_location.split()).lower():
+            raise CLIError(
+                "Replication should not be in the same location '{}' as the registry."
```

Comment:
I might end up removing this validation. The difference is whether we want to fail the request quickly on the client side or send the request to RP and then fail.